src/modules.mjs

```diff
@@ -377,8 +377,8 @@ export class SourceTextModuleRecord extends CyclicModuleRecord {
       // a. Let importedModule be ! HostResolveImportedModule(module, in.[[ModuleRequest]]).
       const importedModule = X(HostResolveImportedModule(module, ie.ModuleRequest));
       // b. NOTE: The above call cannot fail because imported module requests are a subset of module.[[RequestedModules]], and these have been resolved earlier in this algorithm.
-      // c. If in.[[ImportName]] is ~star~, then
-      if (ie.ImportName === 'star') {
+      // c. If in.[[ImportName]] is ~namespace-object~, then
+      if (ie.ImportName === 'namespace-object') {
```

Comment:
These aren't just strings, they are spec enum values. They are in an untagged union with all strings, meaning you should have some way to distinguish them from the string with similar contents. I'm not 100% sure how engine262 works, but it looks like you're not making that distinction here (nor was it being made before).

Reply:
@michaelficarra all values in engine262 are instances of (various subclasses of) the `Value` class, so they can't be confused with values in the host. https://github.com/engine262/engine262/blob/main/src/value.mjs#L79

Reply:
Okay, cool.